### PR TITLE
✨ Improve webhook loggings

### PR DESCRIPTION
**Pull request by anthonyho007**
I have added more logs at places where it deems necessary. If you guys have any good ideas, please let me know!

Issue: https://github.com/kubernetes-sigs/controller-runtime/issues/340

### CODE REVIEW

File: pkg/webhook/admission/http.go [hunk 1]
```diff
@@ -97,5 +97,7 @@ func (wh *Webhook) writeResponse(w io.Writer, response Response) {
 	if err != nil {
 		wh.log.Error(err, "unable to encode the response")
 		wh.writeResponse(w, Errored(http.StatusInternalServerError, err))
+	} else {
+		wh.log.V(1).Info("encoded response", "admission response", responseAdmissionReview)
```
Comment (mengqiy) [inline pkg/webhook/admission/http.go]: responseAdmissionReview is not encoded.
```suggestions
wh.log.V(1).Info("admission response", "response", responseAdmissionReview)
```

File: pkg/webhook/server.go [hunk 2]
```diff
@@ -171,13 +174,15 @@ func (s *Server) Start(stop <-chan struct{}) error {
 			// Error from closing listeners, or context timeout
 			log.Error(err, "error shutting down the HTTP server")
 		}
+		log.Info("shutting down webhook server")
```
[the remark this answers is not in the document]
Comment (anthonyho007) [reply pkg/webhook/server.go]: done

File: pkg/webhook/server.go [hunk 3]
```diff
@@ -171,13 +174,15 @@ func (s *Server) Start(stop <-chan struct{}) error {
...
 		close(idleConnsClosed)
 	}()
 
 	err = srv.Serve(listener)
 	if err != nil && err != http.ErrServerClosed {
 		return err
 	}
+	log.Info("webhook server is serving at ", "host", s.Host, "port", s.Port)
```
[the remark this answers is not in the document]
Comment (anthonyho007) [reply pkg/webhook/server.go]: done

File: pkg/webhook/admission/http.go (hunk 1 above)
Comment (anthonyho007) [reply pkg/webhook/admission/http.go]: done

File: pkg/webhook/server.go [hunk 4]
```diff
@@ -132,6 +134,7 @@ func (s *Server) Start(stop <-chan struct{}) error {
 		if _, err := inject.LoggerInto(baseHookLog.WithValues("webhook", hookPath), webhook); err != nil {
 			return err
 		}
+		baseHookLog.Info("loaded webhook in", "webhook path", hookPath)
```
Comment (mengqiy) [inline pkg/webhook/server.go]: IMO we should put it in https://github.com/kubernetes-sigs/controller-runtime/blob/13ee2bc1b81d8c3ef01ea48d28c4028933865e5b/pkg/webhook/server.go#L100
```
baseHookLog.Info("registering webhook", "path", path)
s.webhooks[path] = hook
s.WebhookMux.Handle(path, instrumentedHook(path, hook))
```

File: pkg/webhook/server.go [hunk 5]
```diff
@@ -158,13 +161,16 @@ func (s *Server) Start(stop <-chan struct{}) error {
 		return err
 	}
 
+	log.Info("webhook server is serving at ", "host", s.Host, "port", s.Port)
```
Comment (mengqiy) [inline pkg/webhook/server.go]: Usually `msg` should be a complete sentence.
e.g. `log.Info("webhook server is serving", "host", s.Host, "port", s.Port)`

Ref: https://github.com/kubernetes-sigs/controller-runtime/blob/master/TMP-LOGGING.md

Comment (anthonyho007) [reply pkg/webhook/server.go]: done

**Comment by anthonyho007** [comment]
Hi @DirectXMan12! Instead of logging the entire admissionResponse and admissionRequest, i have selected a couple fields to be included in the logs. Plz let me know if this is okay 😃 !

**Comment by DirectXMan12** [comment]
that looks a lot better

/lgtm
/approve

**Comment by mengqiy** [comment]
@anthonyho007 can you please fix the failing tests?

**Pull request merged**